### PR TITLE
Added linting and auditing to our CI pipeline

### LINKED ISSUE #39: Linting
We're already enforcing format constraints, might as well add linting too!

### PR DESCRIPTION
**Pull request by chanson02**
@abrahimt we can both work on trying to get this to pass.

### CODE REVIEW

File: .github/workflows/rust.yml [hunk 1]
```diff
@@ -17,6 +17,17 @@ jobs:
       - name: Format
         run: cargo fmt -- --check
 
+  audit:
```
Comment (chanson02) [inline .github/workflows/rust.yml:20]: Change to lint?

Comment (abrahimt) [reply .github/workflows/rust.yml:20]: I think we add it

Comment (chanson02) [reply .github/workflows/rust.yml:20]: Well I thought they were similar enough they could go in the same job, but I wasn't sure what to name that job lol

File: src/main.rs [hunk 2]
```diff
@@ -26,13 +26,13 @@ fn main() {
...
+// //min function
+// #[rustfmt::skip]
+// fn min<T: Ord>(a: T, b: T) -> T {
+//     if a < b { a } else { b }
+// }
+// //max function
+// #[rustfmt::skip]
+// fn max<T: Ord>(a: T, b: T) -> T {
+//     if a > b { a } else { b }
+// }
```
Comment (chanson02) [inline src/main.rs:38]: We can use #[rustfmt::skip] to skip formatting, I wonder if we can use something simialr to skip linting with clippy.

File: src/fen_util.rs [hunk 3]
```diff
@@ -47,6 +47,7 @@ pub fn place_pieces(chessboard: &mut Chessboard, fen_rows: &str) {
 /// # Returns
 ///
 /// A boolean indicating whether the FEN string is valid.
+#[allow(unused_variables)]
```
Comment (chanson02) [inline src/fen_util.rs:50]: @abrahimt what's this for?

Comment (abrahimt) [reply src/fen_util.rs:50]: I think I put that during dev and forgot to take it out. It shouldn't be there.

**Pull request merged**